sandy/core/cov.py

```diff
@@ -831,11 +822,11 @@ def sampling(self, nsmp, seed=None, rows=None, pdf='normal',
             samples = np.exp(samples.add(self.log2norm_mean(ones), axis=0))
         elif relative:
             samples += 1
-        if truncate_:
-            lower_bound = samples > 0
-            upper_bound = samples < 2
-            samples = samples.where(lower_bound, 0)
-            samples = samples.where(upper_bound, 2)
+            if not already_cut:

```

Comment:
Do we need `already_cut`? 
In principle you can always apply this truncation, independently whether `already_cut=True` or `already_cut=False`. The results should be the same.
I am not against it, but if it useless (or if it saves only a marginal amount of computational time) then I would remove it to simplify the code.